compiler/rustc_data_structures/src/thin_vec.rs

```diff
@@ -10,6 +12,14 @@ impl<T> ThinVec<T> {
     pub fn new() -> Self {
         ThinVec(None)
     }
+
+    pub fn iter(&self) -> std::slice::Iter<'_, T> {
+        self.into_iter()
+    }
+
+    pub fn iter_mut(&mut self) -> std::slice::IterMut<'_, T> {
+        self.into_iter()
+    }
 }
```

Comment:
It looks like this is also missing `.push` and `.pop`, I can add those separately. Seems weird that this derefs to a slice instead of a vec.

Reply:
You couldn't `DerefMut` to a vec in the None case.